src/prefect/infrastructure/docker.py

```diff
@@ -162,18 +169,39 @@ class DockerContainer(Infrastructure):
     necessary and the API is connectable while bound to localhost.
     """
 
-    image: str = Field(default_factory=get_prefect_image_name)
-    image_pull_policy: ImagePullPolicy = None
+    type: Literal["docker-container"] = Field(
+        "docker-container", description="The type of infrastructure"
+    )
```

Comment:
Does this slug exist in the block model now? Do we need a separate field?

Reply:
I was thinking the same thing while working on this. I think the block type slug can replace `type` on the infrastructure blocks. I'll open an issue to track removing `type`.